backend/pkg/api/handlers/common.go

```diff
@@ -337,6 +336,30 @@ func (h *HandlerService) handleDashboardId(ctx context.Context, param string) (*
 	return dashboardId, nil
 }
 
+const chartDatapointLimit uint64 = 200
+
+// helper function to retrieve allowed chart timestamp boundaries according to the users premium perks at the current point in time
+func (h *HandlerService) getCurrentChartTimeLimitsForUser(ctx context.Context, dashboardId *types.VDBId, aggregation enums.ChartAggregation) (uint64, uint64, uint64, error) {
```

Comment:
Minor: For clarity
```suggestion
func (h *HandlerService) getCurrentChartTimeLimitsForDashboard(ctx context.Context, dashboardId *types.VDBId, aggregation enums.ChartAggregation) (uint64, uint64, uint64, error) {
```
Since this is dependent on the dashboard owner (i.e. in case of a shared dasboard), not the requesting user.